source/features/view-markdown-source.tsx

```diff
@@ -1,130 +1,33 @@
-*/
-async function showSource(): Promise<void> {
-	const sourceButton = select('button.rgh-md-source')!;
-	const renderedButton = select('button.rgh-md-rendered')!;
-
-	sourceButton.disabled = true;
-
-	const source = buttonBodyMap.get(sourceButton) ?? fetchSource();
-	const rendered = await buttonBodyMap.get(renderedButton) ?? select('.blob.js-code-block-container')!;
-
-	buttonBodyMap.set(sourceButton, source);
-	buttonBodyMap.set(renderedButton, rendered);
-
-	rendered.replaceWith(await source);
-
-	sourceButton.disabled = false;
-
-	sourceButton.classList.add('selected');
-	renderedButton.classList.remove('selected');
-	blurButton(sourceButton);
-	dispatchEvent(sourceButton, 'rgh:view-markdown-source');
-
-	(await source).before(await lineActions());
-}
-
-async function showRendered(): Promise<void> {
-	const sourceButton = select('button.rgh-md-source')!;
-	const renderedButton = select('button.rgh-md-rendered')!;
-
-	renderedButton.disabled = true;
-
-	(await buttonBodyMap.get(sourceButton))!.replaceWith(await buttonBodyMap.get(renderedButton)!);
-
-	renderedButton.disabled = false;
-
-	sourceButton.classList.remove('selected');
-	renderedButton.classList.add('selected');
-	blurButton(renderedButton);
-	dispatchEvent(sourceButton, 'rgh:view-markdown-rendered');
-
-	(await lineActions()).remove();
-}
 
 async function init(): Promise<void> {
-	delegate(document, '.rgh-md-source:not(.selected)', 'click', showSource);
-	delegate(document, '.rgh-md-rendered:not(.selected)', 'click', showRendered);
-
+	const isPlain = new URLSearchParams(location.hash).get('plain') === '1';
 	select('#raw-url')!.closest('.d-flex')!.prepend(
-		<div className="BtnGroup">
-			<button className="btn btn-sm BtnGroup-item tooltipped tooltipped tooltipped-nw rgh-md-source" type="button" aria-label="Display the source blob">
+		<div className="BtnGroup rgh-view-markdown-source">
+			<a href="?plain=1" className="btn btn-sm BtnGroup-item tooltipped tooltipped-nw" aria-label="Display the source">
 				<CodeIcon/>
-			</button>
-			<button className="btn btn-sm BtnGroup-item tooltipped tooltipped-nw rgh-md-rendered selected" type="button" aria-label="Display the rendered blob">
+			</a>
+			<a href={location.pathname} className="btn btn-sm BtnGroup-item tooltipped tooltipped-nw" aria-label="Display the rendered file">
 				<FileIcon/>
-			</button>
+			</a>
 		</div>
 	);
 
-	// Add support for permalinks to the code
-	if (location.hash.startsWith('#L')) {
-		await showSource();
-
-		// Enable selected line highlight
-		window.dispatchEvent(new HashChangeEvent('hashchange', {
-			oldURL: location.href,
-			newURL: location.href
-		}));
-	}
+	select('rgh-view-markdown-source').children[isPlain ? 1 : 0].classList.add('selected');
 }
 
 void features.add(__filebasename, {
 	include: [
 		pageDetect.isSingleFile
 	],
 	exclude: [
-		() => !select.exists('.blob .markdown-body')
+		() => !select.exists('.blob .markdown-body, .type-markdown')
```

Comment:
If it only works with markdown files for now, we should just use a single isMarkdownFile `include`. We have it somewhere in our code

Reply:
I couldn't find `isMarkdownFile`, but how about this?
```tsx
  include: [
  () => location.pathname.endsWith('.md')
  ],
```

Reply:
That’s not enough, markdown has plenty of extensions. You can copy/adapt the regex from this file. Please don’t change that file: 

https://github.com/sindresorhus/refined-github/blob/44f4ddc09618b2dadea8ff1a7aa19380708f2e1a/source/features/vertical-front-matter.tsx#L9

Reply:
I mean that's what I copied from here:
https://github.com/sindresorhus/refined-github/blob/44f4ddc09618b2dadea8ff1a7aa19380708f2e1a/source/features/hide-markdown-diff.tsx#L37
No problem though, I'm on it! :)